### PR TITLE
add Restaurants dataset

### PR DESCRIPTION
**Pull request by EbrahimKreem**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. 
If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to static food data?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by EbrahimKreem** [comment]
I added this dataset because it contains small headers and it will make testing easier

**Comment by EbrahimKreem** [comment]
the next change I want to add is `write-json-file` package that will make writing JSON files to the filesystem easier

**Comment by atherdon** [comment]
1) Is restaurants is part of FAO?  if no, it's better to create a 4th directory and 
2) why you don't like a generator module that we created? I mean I'm ok if we have a better solution, but we work on that very hard and we tested it frequently with our cases. I mean for me now it just delaying the whole finishing part again.
So you'll need to convince me that this module are really better, not just your wish to use something new :)

i understand that our generator is not ideal, for sure, because it was done by interns, but we tested it at a lot of different cases - and it was battle-tested and improved few times. So i'm not sure. only if you spend a lot of time - rewrite whole stuff with your new module - then I'll be happy :)

**Comment by EbrahimKreem** [comment]
:laughing: I know I am the laziest developer ever but your comments make me in right direction :+1:

**Comment by atherdon** [comment]
Haha - don't worry about it. I think you have nice skills but you still not
open them to me :) maybe those tasks are boring for you, not sure. But you
know - work is work and 80% it's about to make things done.

i'm totally fine with that, just want to share my point of view.

i mean i'm sure that we can just spend 2-5hours and everything will work
well for projects2.0 because i literally removed a lot of duplication few
days ago. we just need to overcome the last bumps

>

**Comment by EbrahimKreem** [comment]
I read the code of `assign` and `generate` in `csvToJson.js` file but didn't know what they are trying to achieve it will be helpful if there is a former intern who writes the code to explain what those functions do if I want to generate JSON file I will make the `csvToJson` fn looks like this:

Screenshot from 2019-10-07 13-25-07
 because in my opinion if I am trying to generate JSON file I will make a function like this:

**Comment by EbrahimKreem** [comment]
> Can you compare logic of parser that you've created with other files?
I mean I updated the code, so I think this file should have a next version as well

ok

**Comment by EbrahimKreem** [comment]
I tested the function and it works correctly I need to know what should `assign` and `generate` do?

**Comment by atherdon** [comment]
i think some code was done by @vadim9999 but if he is busy - you'll need to debug it.
but we don't need to update a lot of code inside, because as you might remember - it was working before. I think the biggest issue now - is to figure out how to pass paths and then everything will work for most cases/csv files

**Comment by atherdon** [comment]
> I tested the function and it works correctly I need to know what should `assign` and `generate` do?

are those methods from generator module?

**Comment by EbrahimKreem** [comment]
with the latest commit, it creates a JSON file successfully in the directory

**Comment by EbrahimKreem** [comment]
> are those methods from generator module?
no

**Comment by atherdon** [comment]
> > are those methods from generator module?

 > no

I think you should check this method. I'm not sure, but it was the latest version by my opinion
https://github.com/GroceriStar/food-datasets-csv-parser/blob/master/src/fileSystem.js#L31

**Comment by EbrahimKreem** [comment]
I checked this method and some of the code in the new `csvToJson` is taken from `parseDirectoryFiles`

**Comment by atherdon** [comment]
1) https://github.com/GroceriStar/food-datasets-csv-parser#assign-filename-dataentries-
2) I don't like that instead of using `write`from generator we're using fs method

**Comment by atherdon** [comment]
i think csvToJson should be improved a bit. 
one of our goal is to have inside of csvtojson 2 options:
1) you parsing whole CSV file into one json file
2) you parsing CSV file into a set of json files, that splitted in chunks.

for second case I think we should use assign, that has `write` inside of it

Btw, can you update other parsers accordingly and test how they works with changes that youve made?

**Comment by atherdon** [comment]
https://github.com/GroceriStar/food-static-files-generator/blob/master/package/src/fileSystem.js#L34

**Comment by EbrahimKreem** [comment]
> you parsing CSV file into a set of json files, that splitted in chunks.

that is confusing why it should be splitted in chunks? what are the benefits of that?

**Comment by atherdon** [comment]
the same reason why we need a separated generator module - it's part of our
project :)
and usually - default datasets not enough, so later we'll merge data
together or change logic.
example - we got an FAO dataset. it have a lot of information, but imagine
that we don't need all information, so we change number of fields that we
have

**Comment by atherdon** [comment]
tell me when I should merge this pull request

**Comment by EbrahimKreem** [comment]
> the same reason why we need a separated generator module - it's part of our
project :)
and usually - default datasets not enough, so later we'll merge data
together or change logic.
example - we got an FAO dataset. it have a lot of information, but imagine
that we don't need all information, so we change number of fields that we
have

sorry but it doesn't click in my head but I think with testing datasets it will be understood

**Comment by EbrahimKreem** [comment]
> tell me when I should merge this pull request

you can merge it any time you want I think I completed all your pending comments

**Comment by EbrahimKreem** [comment]
what are the pending tasks that should be resolved to complete this module

**Comment by atherdon** [comment]
https://github.com/GroceriStar/food-datasets-csv-parser/pull/191#issuecomment-538945865

**Comment by EbrahimKreem** [comment]
new Guinness  world record  :laughing: 
Screenshot from 2019-10-08 22-26-04

**Comment by atherdon** [comment]
it's great! keep going

**Comment by EbrahimKreem** [comment]
okay I completed the pending comments I think you should merge this PR

**Comment by atherdon** [comment]
reviewing

File: src/csvToJson.js [hunk 1]
```diff
@@ -45,13 +46,30 @@ const assign = (file, dataEntries) => {
...
+ * @async
+ * @param {dirPath} dirPath directory path
+ * @param {data} data
+ * @returns {Promise<void>} Promise
+ */
+const csvToJson = (dirPath, data) => {
+  // stringify data with indent
+  const json = JSON.stringify(data, null, 2);
+
+  fs.readdir(dirPath, async (err, files) => {
```
Comment (atherdon) [inline src/csvToJson.js]: Can we replace this method with `read` method from generator?

Comment (EbrahimKreem) [reply src/csvToJson.js]: which method `fs.readdir` or `JSON.stringify`

**Pull request merged**